src/parser/contexts.h

```diff
@@ -684,6 +687,31 @@ struct ParseDeclsCtx : NullTypeParserCtx, NullInstrParserCtx {
                   ImportNames* import,
                   TypeUseT type,
                   Index pos);
+
+  Result<> addFuncExport(Index pos, FuncIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
+
+  Result<> addTableExport(Index pos, TableIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
+
+  Result<> addMemoryExport(Index pos, MemoryIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
+
+  Result<> addGlobalExport(Index pos, GlobalIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
+
+  Result<> addTagExport(Index pos, TagIdxT, Name) {
+    exportDefs.push_back(pos);
+    return Ok{};
+  }
```

Comment:
Can we use a template over these 5 on the second param?

Reply:
Hmm yeah I'll try that out.

Reply:
Ok, there's only a single `addExport` method on each context now.